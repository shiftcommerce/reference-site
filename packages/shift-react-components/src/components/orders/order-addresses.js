// Libraries
import React, { PureComponent } from 'react'

export class OrderAddresses extends PureComponent {
  render () {
    return (
      <>
        { this.props.addresses.map((address) => {
          return (
            <div className='c-order-history__address' key={address.id}>
              <p className='u-bold'>{ address.name }</p>
              <p>{ address.company }</p>
              { address.lines.map((line => { return <p>{line}</p> })) }
              <p>{ address.city }</p>
              <p>{ address.state }</p>
              <p>{ address.postcode }</p>
              <p>{ address.country }</p>
            </div>
          )
        })}
      </>
    )
  }
}
