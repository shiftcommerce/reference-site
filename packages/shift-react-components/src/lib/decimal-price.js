function decimalPrice (price) {
  return (price).toFixed(2)
}

export default decimalPrice
