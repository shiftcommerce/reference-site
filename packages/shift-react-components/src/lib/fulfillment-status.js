/**
 * Converts the OMS status to a user friendly one
 * @param  {String} status
 * @return {String}
 */
export const fulfillmentStatus = (status) => {
  const mappedStatuses = {
    pending_approval: 'Awaiting shipment',
    unfulfilled: 'Awaiting shipment',
    short_shipped: 'Short shipped',
    partially_shipped: 'Partially shipped',
    fully_shipped: 'Shipped'
  }

  return mappedStatuses[status]
}
