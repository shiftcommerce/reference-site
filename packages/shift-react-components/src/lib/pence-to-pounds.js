/**
 * Convert the price from pence to pounds. For example:
 * 2547 pence becomes 25.47
 * @param  {Number} priceInPence
 * @return {Number} - The price
 */

function penceToPounds (priceInPence) {
  return (priceInPence / 100).toFixed(2)
}

export default penceToPounds
