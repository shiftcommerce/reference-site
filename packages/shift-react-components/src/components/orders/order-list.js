// Libraries
import React, { PureComponent, Fragment } from 'react'
import format from 'date-fns/format'
import t from 'typy'

// Lib
import { fulfillmentStatus } from '../../lib/fulfillment-status'
import { penceToPounds } from '../../lib/pence-to-pounds'

export class OrderList extends PureComponent {
  constructor (props) {
    super(props)

    this.headers = props.headers || ['Order No.', 'Order Date', 'Cost', 'Shipping Status', '']
  }

  /**
  * Render the order details
  * @param  {Object} order
  * @param  {Object} orderDate
  * @param  {Object} total
  * @return {string} - HTML markup for the component
  */
  renderOrderDetails (order) {
    const { updateCurrentOrder } = this.props

    const orderDate = format(new Date(order.placed_at), 'D MMM YYYY')
    const orderTotal = penceToPounds(t(order, 'pricing.total_inc_tax').safeObject)

    return [
      {
        label: this.headers[0],
        value: order.reference
      },
      {
        label: this.headers[1],
        value: orderDate
      },
      {
        label: this.headers[2],
        value: `${order.pricing.currency}${orderTotal}`
      },
      {
        label: this.headers[3],
        value: fulfillmentStatus(order.fulfillment_status)
      },
      {
        label: this.headers[4],
        value: <a href={`/account/orders/${order.reference}`} className='o-button o-button--primary' onClick={(event) => updateCurrentOrder(event, order.reference)}>View Details</a>
      }
    ].map((column, columnIndex) => {
      return (
        <div
          key={columnIndex}
          className='c-order-history-table__col'
        >
          { column.label && <span className='c-order-history-table__label'>{ column.label }</span> }
          { column.value }
        </div>
      )
    })
  }

  /**
  * Render the header for the order list
  * @param  {Object} order
  * @param  {Object} orderDate
  * @param  {Object} total
  * @return {string} - HTML markup for the component
  */
  renderOrderHeader (order, orderDate, total) {
    return (
      <Fragment>
        <h4>{ orderDate } - { order.reference } - Â£{ total }</h4>
        <label htmlFor={order.reference} className='c-order-history__header-label' />
        <input type='checkbox' id={order.reference} />
        <span className='c-order-history__arrow' />
      </Fragment>
    )
  }

  renderTableHeader () {
    return this.headers.map((column, columnIndex) => {
      return (
        <div
          key={columnIndex}
          className='c-order-history-table__col c-order-history-table__col--header'
        >
          { column }
        </div>
      )
    })
  }

  render () {
    const { orders } = this.props

    return (
      <div className='c-order-history-table'>
        <div className='c-order-history-table__row c-order-history-table__row--header'>
          { this.renderTableHeader() }
        </div>
        { orders.data.map((order) => {
          return (
            <div
              key={order.id}
              className='c-order-history-table__row c-order-history-table__row--body'
            >
              { this.renderOrderDetails(order) }
            </div>
          )
        }) }
      </div>
    )
  }
}
