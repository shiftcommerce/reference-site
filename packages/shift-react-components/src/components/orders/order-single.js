// Libraries
import React, { PureComponent, Fragment } from 'react'
import format from 'date-fns/format'

// Lib
import { penceToPounds } from '../../lib/pence-to-pounds'

// Components
import ShippingAddresses from './shipping-addresses'
import { LineItems } from '../cart/line-items'

import Link from '../../objects/link'

export class OrderSingle extends PureComponent {
  /**
   * Calculate the subtotal based on each line item price
   * @param  {Object} order
   * @return {Number}
   */
  calculateSubTotal (order) {
    return order.line_items.reduce(
      (accumulator, currentValue) => accumulator + currentValue.pricing.line_total_inc_tax, 0
    )
  }

  /**
   * Render the total shipping costs in the order list
   * @param  {Object} order
   * @return {string} - HTML markup for the component
   */
  calculateShippingTotal (order) {
    return order.shipping_methods.reduce(
      (accumulator, currentValue) => accumulator + currentValue.price, 0
    )
  }

  /**
   * Get the tax amount based on the totals
   * @param  {Object} order
   * @return {Number}
   */
  calculateTax (order) {
    return order.pricing.total_inc_tax - order.pricing.total_ex_tax
  }

  /**
   * Reformat the line items to they'll work with the <LineItems /> component
   * @param  {Array} lineItems
   * @return {Array}
   */
  formatLineItems (lineItems) {
    return lineItems.map((lineItem, lineItemIndex) => {
      return {
        id: lineItemIndex,
        item: {
          sku: lineItem.sku,
          title: 'item title',
          picture_url: lineItem.image_urls[0],
          product: {
            title: 'product title',
            slug: 'slug'
          }
        },
        unit_quantity: lineItem.quantity,
        total: (lineItem.pricing.line_total_inc_tax / 100),
        sub_total: (lineItem.pricing.line_total_inc_tax / 100)
      }
    })
  }

  /**
   * Render each part of the order items
   * @param  {String}                 options.header
   * @param  {String|Number|Element0} options.body
   * @return {String} - HTML markup for the component
   */
  renderItem ({ header, body }) {
    return (
      <Fragment>
        { header && <h5 className='c-order-single__header'>{ header }</h5> }
        { body && <div className='c-order-single__body'>{ body }</div> }
      </Fragment>
    )
  }

  render () {
    const { order } = this.props
    const lineItems = this.formatLineItems(order.line_items)

    const items = [{
      header: 'Order No.',
      body: order.reference
    }, {
      header: 'Order Date',
      body: format(new Date(order.placed_at), 'MMM D, YYYY')
    }, {
      header: 'Cost',
      body: `&pound;${penceToPounds(order.pricing.total_inc_tax)}`
    }, {
      header: 'Shipping Method',
      body: order.shipping_methods[0].label
    }, {
      header: 'Billing Address',
      body: <ShippingAddresses addresses={order.shipping_addresses} />
    }, {
      header: 'Shipping Address',
      body: <ShippingAddresses addresses={order.shipping_addresses} />
    }, {
      header: 'Currency',
      body: order.pricing.currency
    }]

    return (
      <Fragment>
        <div className='c-order-history__nav'>
          <h2>Order { order.reference }</h2>
        </div>
        <div className='c-order-single'>
          { items.map((item, itemIndex) => {
            return (
              <div key={itemIndex} className='c-order-single__item'>
                { this.renderItem(item) }
              </div>
            )
          }) }
          <div className='c-order-single__item c-order-single__item--full'>
            <h5 className='c-order-single__header'>Order Summary</h5>
            <div className='c-order-single__body'>
              <div className='o-grid-container'>
                <div className='o-col-1-7'>
                  <LineItems
                    lineItems={lineItems}
                    lineItemsCount={lineItems.length}
                  />
                </div>
                <div className='o-col-8-13'>
                  <div className='c-cart-summary'>
                    <dl aria-label='Subtotal'>
                      <dt> Subtotal: </dt>
                      <dd> &pound;{ penceToPounds(this.calculateSubTotal(order)) } </dd>
                    </dl>
                    <dl aria-label='Taxes'>
                      <dt> UK VAT 20% (Included in Price): </dt>
                      <dd> &pound;{ penceToPounds(this.calculateTax(order)) } </dd>
                    </dl>
                    <dl aria-label='Shipping'>
                      <dt> Shipping Total: </dt>
                      <dd> &pound;{ penceToPounds(this.calculateShippingTotal(order)) } </dd>
                    </dl>
                    <dl aria-label='Total' className='c-cart-summary__total'>
                      <dt> Total: </dt>
                      <dd> &pound;{ penceToPounds(order.pricing.total_inc_tax) } </dd>
                    </dl>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className='c-order-single__item c-order-single__item--full'>
            <div className='c-order-single__body'>
              <Link href='/account/orders' className='o-button o-button--primary'>View all orders</Link>
            </div>
          </div>

        </div>
      </Fragment>
    )
  }
}
