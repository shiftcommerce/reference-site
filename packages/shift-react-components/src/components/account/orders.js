// Libraries
import React, { Component, Fragment } from 'react'

// Lib
import Config from '../../lib/config'
import { Loading } from '../../objects/loading'
import { OrderList } from '../../components/orders/order-list'
import { OrderSingle } from '../../components/orders/order-single'

export class AccountOrders extends Component {
  constructor (props) {
    super(props)

    this.Link = Config.get().Link
  }

  componentDidMount () {
    this.props.fetchOrders()
  }

  /**
   * Render the list of orders, or the individual order
   * @param  {Object} orders
   * @return {string} - HTML markup for the component
   */
  renderOrders (orders) {
    const {
      currentOrderRef,
      fetchOrders,
      pageNumber,
      pagePath,
      updateCurrentOrder
    } = this.props

    if (orders.data.length === 0) {
      return (<p>No previous orders found.</p>)
    }

    if (currentOrderRef) {
      const currentOrder = orders.data.find((order) => {
        return order.reference === currentOrderRef
      })

      return (
        <OrderSingle
          order={currentOrder}
          updateCurrentOrder={updateCurrentOrder}
        />
      )
    }

    return (
      <Fragment>
        <div className='c-order-history__nav'>
          <h2>Order History</h2>
        </div>
        <OrderList
          orders={orders}
          pageNumber={pageNumber}
          pagePath={pagePath}
          updateCurrentOrder={updateCurrentOrder}
        />
      </Fragment>
    )
  }

  render () {
    const { orders, orders: { loading } } = this.props

    return (
      <div className='c-order-history'>
        { loading ? <Loading /> : this.renderOrders(orders) }
      </div>
    )
  }
}
