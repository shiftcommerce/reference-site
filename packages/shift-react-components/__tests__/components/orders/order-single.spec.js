// Libraries
import React from 'react'
import format from 'date-fns/format'

// Component
import { OrderSingle } from '../../../src/components/orders/order-single'

// Lib
import { penceToPounds } from '../../../src/lib/pence-to-pounds'

// Fixtures
import orders from '../../fixtures/orders'

test('renders correctly', () => {
  // Act
  const wrapper = mount(
    <OrderSingle order={orders.data[0]} />
  )

  const instance = wrapper.instance()

  const order = orders.data[0]
  const shippingTotal = penceToPounds(instance.calculateShippingTotal(order))
  const orderSubTotal = penceToPounds(instance.calculateSubTotal(order))
  const orderTotal = penceToPounds(order.pricing.total_inc_tax)
  const orderDate = format(new Date(order.placed_at), 'MMM D, YYYY')

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(shippingTotal)
  expect(wrapper).toIncludeText(orderSubTotal)
  expect(wrapper).toIncludeText(orderTotal)
  expect(wrapper).toIncludeText(orderDate)
})
