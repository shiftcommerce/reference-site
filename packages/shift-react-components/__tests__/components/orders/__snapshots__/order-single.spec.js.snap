// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<OrderSingle
  order={
    Object {
      "account_reference": "shiftacc",
      "customer": Object {
        "email": "test@shiftcommerce.com",
        "id": "61c40964-4b4c-4f54-b580-958798087312",
        "meta_attributes": Object {},
        "name": "SHIFT",
        "reference": "23063301",
      },
      "id": "61c40964-4b4c-4f54-b580-958798087312",
      "line_items": Array [
        Object {
          "discounts": Array [],
          "id": "61c40964-4b4c-4f54-b580-958798087312_0",
          "image_urls": Array [
            "https://example.com/path/to/image.jpg",
            "https://example.com/path/to/image1.jpg",
          ],
          "individual_prices": Array [],
          "pricing": Object {
            "each_ex_tax": 1200,
            "each_inc_tax": 1440,
            "line_discount_ex_tax": 0,
            "line_discount_inc_tax": 0,
            "line_ex_tax": 6000,
            "line_total_inc_tax": 7200,
            "tax_rate": 12000,
          },
          "quantity": 5,
          "shipping_address": Object {
            "city": "Leeds",
            "company": "SHIFT Commerce",
            "country": "United Kingdom",
            "id": "61c40964-4b4c-4f54-b580-958798087312_1",
            "lines": Array [
              "The Calls",
            ],
            "meta_attributes": Object {
              "further": "information",
            },
            "name": "Test Customer",
            "postcode": "LS2 7EY",
            "state": "West Yorkshire",
          },
          "shipping_method": Object {
            "id": "61c40964-4b4c-4f54-b580-958798087312_1",
            "label": "Next Day Delivery",
            "meta_attributes": Object {
              "sku": "NEXTDAY",
            },
            "price": 599,
          },
          "sku": "WHITE-TSHIRT-XXL",
        },
        Object {
          "discounts": Array [
            Object {
              "amount_ex_tax": 200,
              "amount_inc_tax": 240,
              "coupon_code": "3FOR2SOCKS",
              "id": "61c40964-4b4c-4f54-b580-958798087312_1_0",
              "label": "3 for 2 on Socks",
              "meta_attributes": Object {},
            },
          ],
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "image_urls": Array [],
          "individual_prices": Array [
            Object {
              "discount_ex_tax": 67,
              "discount_inc_tax": 80,
              "discounts": Array [
                Object {
                  "amount_ex_tax": 67,
                  "amount_inc_tax": 80,
                  "coupon_code": null,
                  "id": "61c40964-4b4c-4f54-b580-958798087312_1_0_0",
                  "label": "3 for 2 on Socks",
                  "meta_attributes": Object {},
                },
              ],
              "ex_tax": 133,
              "id": "61c40964-4b4c-4f54-b580-958798087312_1_0",
              "inc_tax": 160,
            },
            Object {
              "discount_ex_tax": 67,
              "discount_inc_tax": 80,
              "discounts": Array [
                Object {
                  "amount_ex_tax": 67,
                  "amount_inc_tax": 80,
                  "coupon_code": null,
                  "id": "61c40964-4b4c-4f54-b580-958798087312_1_1_0",
                  "label": "3 for 2 on Socks",
                  "meta_attributes": Object {},
                },
              ],
              "ex_tax": 133,
              "id": "61c40964-4b4c-4f54-b580-958798087312_1_1",
              "inc_tax": 160,
            },
            Object {
              "discount_ex_tax": 66,
              "discount_inc_tax": 80,
              "discounts": Array [
                Object {
                  "amount_ex_tax": 66,
                  "amount_inc_tax": 80,
                  "coupon_code": null,
                  "id": "61c40964-4b4c-4f54-b580-958798087312_1_2_0",
                  "label": "3 for 2 on Socks",
                  "meta_attributes": Object {},
                },
              ],
              "ex_tax": 134,
              "id": "61c40964-4b4c-4f54-b580-958798087312_1_2",
              "inc_tax": 160,
            },
          ],
          "pricing": Object {
            "each_ex_tax": 200,
            "each_inc_tax": 240,
            "line_discount_ex_tax": 200,
            "line_discount_inc_tax": 240,
            "line_ex_tax": 400,
            "line_total_inc_tax": 480,
            "tax_rate": 12000,
          },
          "quantity": 3,
          "shipping_address": Object {
            "city": "Leeds",
            "company": null,
            "country": "United Kingdom",
            "id": "61c40964-4b4c-4f54-b580-958798087312_2",
            "lines": Array [
              "SHIFT Commerce",
              "The Calls Landing",
            ],
            "meta_attributes": Object {
              "further": "information",
            },
            "name": "Test Customer 2",
            "postcode": "LS2 7EY",
            "state": "West Yorkshire",
          },
          "shipping_method": Object {
            "id": "61c40964-4b4c-4f54-b580-958798087312_2",
            "label": "Two-man Delivery",
            "meta_attributes": Object {
              "sku": "DROPSHIP",
            },
            "price": 2499,
          },
          "sku": "BLACK-SOCKS-001",
        },
        Object {
          "discounts": Array [],
          "id": "61c40964-4b4c-4f54-b580-958798087312_2",
          "image_urls": Array [],
          "individual_prices": Array [],
          "pricing": Object {
            "each_ex_tax": 120000,
            "each_inc_tax": 144000,
            "line_discount_ex_tax": 0,
            "line_discount_inc_tax": 0,
            "line_ex_tax": 120000,
            "line_total_inc_tax": 144000,
            "tax_rate": 12000,
          },
          "quantity": 1,
          "shipping_address": Object {
            "city": "Leeds",
            "company": "SHIFT Commerce",
            "country": "United Kingdom",
            "id": "61c40964-4b4c-4f54-b580-958798087312_1",
            "lines": Array [
              "The Calls",
            ],
            "meta_attributes": Object {
              "further": "information",
            },
            "name": "Test Customer",
            "postcode": "LS2 7EY",
            "state": "West Yorkshire",
          },
          "shipping_method": Object {
            "id": "61c40964-4b4c-4f54-b580-958798087312_1",
            "label": "Next Day Delivery",
            "meta_attributes": Object {
              "sku": "NEXTDAY",
            },
            "price": 599,
          },
          "sku": "WHITE-TSHIRT-L",
        },
      ],
      "placed_at": "2018-01-01T12:00:00.000Z",
      "pricing": Object {
        "currency": "GBP",
        "pre_discount_total_ex_tax": 126600,
        "pre_discount_total_inc_tax": 151440,
        "total_discount_ex_tax": 700,
        "total_discount_inc_tax": 840,
        "total_ex_tax": 125900,
        "total_inc_tax": 150600,
      },
      "reference": "1000",
      "shipping_addresses": Array [
        Object {
          "city": "Leeds",
          "company": "SHIFT Commerce",
          "country": "United Kingdom",
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "lines": Array [
            "The Calls",
          ],
          "meta_attributes": Object {
            "further": "information",
          },
          "name": "Test Customer",
          "postcode": "LS2 7EY",
          "state": "West Yorkshire",
        },
        Object {
          "city": "Leeds",
          "company": null,
          "country": "United Kingdom",
          "id": "61c40964-4b4c-4f54-b580-958798087312_2",
          "lines": Array [
            "SHIFT Commerce",
            "The Calls Landing",
          ],
          "meta_attributes": Object {
            "further": "information",
          },
          "name": "Test Customer 2",
          "postcode": "LS2 7EY",
          "state": "West Yorkshire",
        },
      ],
      "shipping_methods": Array [
        Object {
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "label": "Next Day Delivery",
          "meta_attributes": Object {
            "sku": "NEXTDAY",
          },
          "price": 599,
        },
        Object {
          "id": "61c40964-4b4c-4f54-b580-958798087312_2",
          "label": "Two-man Delivery",
          "meta_attributes": Object {
            "sku": "DROPSHIP",
          },
          "price": 2499,
        },
      ],
    }
  }
>
  <div
    className="c-order-history__nav"
  >
    <h2>
      Order 
      1000
    </h2>
  </div>
  <div
    className="c-order-single"
  >
    <div
      className="c-order-single__item"
      key="0"
    >
      <h5
        className="c-order-single__header"
      >
        Order No.
      </h5>
      <div
        className="c-order-single__body"
      >
        1000
      </div>
    </div>
    <div
      className="c-order-single__item"
      key="1"
    >
      <h5
        className="c-order-single__header"
      >
        Order Date
      </h5>
      <div
        className="c-order-single__body"
      >
        Jan 1, 2018
      </div>
    </div>
    <div
      className="c-order-single__item"
      key="2"
    >
      <h5
        className="c-order-single__header"
      >
        Cost
      </h5>
      <div
        className="c-order-single__body"
      >
        Â£1506.00
      </div>
    </div>
    <div
      className="c-order-single__item"
      key="3"
    >
      <h5
        className="c-order-single__header"
      >
        Shipping Method
      </h5>
      <div
        className="c-order-single__body"
      >
        Next Day Delivery
      </div>
    </div>
    <div
      className="c-order-single__item"
      key="4"
    >
      <h5
        className="c-order-single__header"
      >
        Billing Address
      </h5>
      <div
        className="c-order-single__body"
      >
        <ShippingAddresses
          addresses={
            Array [
              Object {
                "city": "Leeds",
                "company": "SHIFT Commerce",
                "country": "United Kingdom",
                "id": "61c40964-4b4c-4f54-b580-958798087312_1",
                "lines": Array [
                  "The Calls",
                ],
                "meta_attributes": Object {
                  "further": "information",
                },
                "name": "Test Customer",
                "postcode": "LS2 7EY",
                "state": "West Yorkshire",
              },
              Object {
                "city": "Leeds",
                "company": null,
                "country": "United Kingdom",
                "id": "61c40964-4b4c-4f54-b580-958798087312_2",
                "lines": Array [
                  "SHIFT Commerce",
                  "The Calls Landing",
                ],
                "meta_attributes": Object {
                  "further": "information",
                },
                "name": "Test Customer 2",
                "postcode": "LS2 7EY",
                "state": "West Yorkshire",
              },
            ]
          }
        >
          <div
            className="c-order-history__shipping-address"
            key="61c40964-4b4c-4f54-b580-958798087312_1"
          >
            <p
              className="u-bold"
            >
              Test Customer
            </p>
            <p>
              SHIFT Commerce
            </p>
            <p>
              The Calls
            </p>
            <p>
              Leeds
            </p>
            <p>
              West Yorkshire
            </p>
            <p>
              LS2 7EY
            </p>
            <p>
              United Kingdom
            </p>
          </div>
          <div
            className="c-order-history__shipping-address"
            key="61c40964-4b4c-4f54-b580-958798087312_2"
          >
            <p
              className="u-bold"
            >
              Test Customer 2
            </p>
            <p />
            <p>
              SHIFT Commerce,The Calls Landing
            </p>
            <p>
              Leeds
            </p>
            <p>
              West Yorkshire
            </p>
            <p>
              LS2 7EY
            </p>
            <p>
              United Kingdom
            </p>
          </div>
        </ShippingAddresses>
      </div>
    </div>
    <div
      className="c-order-single__item"
      key="5"
    >
      <h5
        className="c-order-single__header"
      >
        Shipping Address
      </h5>
      <div
        className="c-order-single__body"
      >
        <ShippingAddresses
          addresses={
            Array [
              Object {
                "city": "Leeds",
                "company": "SHIFT Commerce",
                "country": "United Kingdom",
                "id": "61c40964-4b4c-4f54-b580-958798087312_1",
                "lines": Array [
                  "The Calls",
                ],
                "meta_attributes": Object {
                  "further": "information",
                },
                "name": "Test Customer",
                "postcode": "LS2 7EY",
                "state": "West Yorkshire",
              },
              Object {
                "city": "Leeds",
                "company": null,
                "country": "United Kingdom",
                "id": "61c40964-4b4c-4f54-b580-958798087312_2",
                "lines": Array [
                  "SHIFT Commerce",
                  "The Calls Landing",
                ],
                "meta_attributes": Object {
                  "further": "information",
                },
                "name": "Test Customer 2",
                "postcode": "LS2 7EY",
                "state": "West Yorkshire",
              },
            ]
          }
        >
          <div
            className="c-order-history__shipping-address"
            key="61c40964-4b4c-4f54-b580-958798087312_1"
          >
            <p
              className="u-bold"
            >
              Test Customer
            </p>
            <p>
              SHIFT Commerce
            </p>
            <p>
              The Calls
            </p>
            <p>
              Leeds
            </p>
            <p>
              West Yorkshire
            </p>
            <p>
              LS2 7EY
            </p>
            <p>
              United Kingdom
            </p>
          </div>
          <div
            className="c-order-history__shipping-address"
            key="61c40964-4b4c-4f54-b580-958798087312_2"
          >
            <p
              className="u-bold"
            >
              Test Customer 2
            </p>
            <p />
            <p>
              SHIFT Commerce,The Calls Landing
            </p>
            <p>
              Leeds
            </p>
            <p>
              West Yorkshire
            </p>
            <p>
              LS2 7EY
            </p>
            <p>
              United Kingdom
            </p>
          </div>
        </ShippingAddresses>
      </div>
    </div>
    <div
      className="c-order-single__item"
      key="6"
    >
      <h5
        className="c-order-single__header"
      >
        Currency
      </h5>
      <div
        className="c-order-single__body"
      >
        GBP
      </div>
    </div>
    <div
      className="c-order-single__item c-order-single__item--full"
    >
      <h5
        className="c-order-single__header"
      >
        Order Summary
      </h5>
      <div
        className="c-order-single__body"
      >
        <div
          className="o-grid-container"
        >
          <div
            className="o-col-1-7"
          >
            <LineItems
              lineItems={
                Array [
                  Object {
                    "id": 0,
                    "item": Object {
                      "picture_url": "https://example.com/path/to/image.jpg",
                      "product": Object {
                        "slug": "slug",
                        "title": "product title",
                      },
                      "sku": "WHITE-TSHIRT-XXL",
                      "title": "WHITE-TSHIRT-XXL",
                    },
                    "sub_total": 72,
                    "total": 72,
                    "unit_quantity": 5,
                  },
                  Object {
                    "id": 1,
                    "item": Object {
                      "picture_url": undefined,
                      "product": Object {
                        "slug": "slug",
                        "title": "product title",
                      },
                      "sku": "BLACK-SOCKS-001",
                      "title": "BLACK-SOCKS-001",
                    },
                    "sub_total": 4.8,
                    "total": 4.8,
                    "unit_quantity": 3,
                  },
                  Object {
                    "id": 2,
                    "item": Object {
                      "picture_url": undefined,
                      "product": Object {
                        "slug": "slug",
                        "title": "product title",
                      },
                      "sku": "WHITE-TSHIRT-L",
                      "title": "WHITE-TSHIRT-L",
                    },
                    "sub_total": 1440,
                    "total": 1440,
                    "unit_quantity": 1,
                  },
                ]
              }
              lineItemsCount={3}
            >
              <div
                className="c-line-items"
              >
                <div
                  className="c-line-items__sections"
                  key="WHITE-TSHIRT-XXL"
                >
                  <div
                    className="c-line-items__images"
                  >
                    <Link
                      href="/slug?slug=undefined"
                    >
                      <a
                        href="/slug?slug=undefined"
                      >
                        <Image
                          alt="WHITE-TSHIRT-XXL"
                          aria-label="WHITE-TSHIRT-XXL"
                          className="c-line-items__image"
                          key="slug"
                          src="https://example.com/path/to/image.jpg"
                        >
                          <div
                            className="c-line-items__image"
                          >
                            <picture>
                              <img
                                alt="WHITE-TSHIRT-XXL"
                                aria-label="WHITE-TSHIRT-XXL"
                                className="o-image"
                                src="https://example.com/path/to/image.jpg"
                                srcSet="https://example.com/path/to/image.jpg 769w"
                              />
                            </picture>
                          </div>
                        </Image>
                      </a>
                    </Link>
                  </div>
                  <div
                    className="c-line-items__information"
                  >
                    <div
                      className="c-line-items__title"
                    >
                      <div
                        className="c-line-items__details"
                      >
                        <h4
                          className="c-line-items__details-title u-bold"
                        >
                          product title - WHITE-TSHIRT-XXL
                        </h4>
                        <div
                          className="c-line-items__details-sku"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                    <div
                      className="c-line-items__quantity"
                    >
                      <div
                        className="c-line-items__quantity-header"
                      >
                        <span>
                          Quantity
                        </span>
                      </div>
                      <span
                        className="c-line-items__quantity-amount"
                      >
                        5
                      </span>
                    </div>
                    <div
                      className="c-line-items__amounts"
                    >
                      <a
                        className="c-line-items__amount c-line-items__amount--total"
                      >
                        Â£
                        72.00
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  className="c-line-items__sections"
                  key="BLACK-SOCKS-001"
                >
                  <div
                    className="c-line-items__images"
                  >
                    <Link
                      href="/slug?slug=undefined"
                    >
                      <a
                        href="/slug?slug=undefined"
                      >
                        <Image
                          alt="BLACK-SOCKS-001"
                          aria-label="BLACK-SOCKS-001"
                          className="c-line-items__image"
                          key="slug"
                        />
                      </a>
                    </Link>
                  </div>
                  <div
                    className="c-line-items__information"
                  >
                    <div
                      className="c-line-items__title"
                    >
                      <div
                        className="c-line-items__details"
                      >
                        <h4
                          className="c-line-items__details-title u-bold"
                        >
                          product title - BLACK-SOCKS-001
                        </h4>
                        <div
                          className="c-line-items__details-sku"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                    <div
                      className="c-line-items__quantity"
                    >
                      <div
                        className="c-line-items__quantity-header"
                      >
                        <span>
                          Quantity
                        </span>
                      </div>
                      <span
                        className="c-line-items__quantity-amount"
                      >
                        3
                      </span>
                    </div>
                    <div
                      className="c-line-items__amounts"
                    >
                      <a
                        className="c-line-items__amount c-line-items__amount--total"
                      >
                        Â£
                        4.80
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  className="c-line-items__sections"
                  key="WHITE-TSHIRT-L"
                >
                  <div
                    className="c-line-items__images"
                  >
                    <Link
                      href="/slug?slug=undefined"
                    >
                      <a
                        href="/slug?slug=undefined"
                      >
                        <Image
                          alt="WHITE-TSHIRT-L"
                          aria-label="WHITE-TSHIRT-L"
                          className="c-line-items__image"
                          key="slug"
                        />
                      </a>
                    </Link>
                  </div>
                  <div
                    className="c-line-items__information"
                  >
                    <div
                      className="c-line-items__title"
                    >
                      <div
                        className="c-line-items__details"
                      >
                        <h4
                          className="c-line-items__details-title u-bold"
                        >
                          product title - WHITE-TSHIRT-L
                        </h4>
                        <div
                          className="c-line-items__details-sku"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                    <div
                      className="c-line-items__quantity"
                    >
                      <div
                        className="c-line-items__quantity-header"
                      >
                        <span>
                          Quantity
                        </span>
                      </div>
                      <span
                        className="c-line-items__quantity-amount"
                      >
                        1
                      </span>
                    </div>
                    <div
                      className="c-line-items__amounts"
                    >
                      <a
                        className="c-line-items__amount c-line-items__amount--total"
                      >
                        Â£
                        1440.00
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </LineItems>
          </div>
          <div
            className="o-col-8-13"
          >
            <div
              className="c-cart-summary"
            >
              <dl
                aria-label="Subtotal"
              >
                <dt>
                   Subtotal: 
                </dt>
                <dd>
                   Â£
                  1516.80
                   
                </dd>
              </dl>
              <dl
                aria-label="Taxes"
              >
                <dt>
                   UK VAT 20% (Included in Price): 
                </dt>
                <dd>
                   Â£
                  247.00
                   
                </dd>
              </dl>
              <dl
                aria-label="Shipping"
              >
                <dt>
                   Shipping Total: 
                </dt>
                <dd>
                   Â£
                  30.98
                   
                </dd>
              </dl>
              <dl
                aria-label="Total"
                className="c-cart-summary__total"
              >
                <dt>
                   Total: 
                </dt>
                <dd>
                   Â£
                  1506.00
                   
                </dd>
              </dl>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c-order-single__item c-order-single__item--full"
    >
      <div
        className="c-order-single__body"
      >
        <a
          className="o-button o-button--primary"
          href="/account/orders"
        >
          View all orders
        </a>
      </div>
    </div>
  </div>
</OrderSingle>
`;
